### PR TITLE
[SourceKit] add qualified types in type requests

### PR DESCRIPTION
**Pull request by skrtks**
This adds the fully qualified type to the response of expression and variable type requests in SourceKit.

The qualified type can be accessed through the `<key.expression_qualified_type>` and `<key.variable_qualified_type>` keys in the response of the expression and variable type requests.

### CODE REVIEW

**Comment by skrtks** [comment]
@ahoppen do you perhaps know who would be the right person to ask as a reviewer here?

**Review by ahoppen — COMMENTED** [review]
Looks good to me in general. A couple conceptual questions:
- What’s the reason you went for a fully-qualified type instead of a USR? Is it because fully-qualified types are easier to parse than USRs?
- I’m thinking whether sending back the fully-qualified types might blow up the response and if we should add an option to the request to control whether fully-qualified types are sent. I don’t have strong opinions either way but @rintaro might.

**Comment by rintaro** [comment]
While it's true that this will more than double the size of the result, since duplicated type names are shared in the string buffer, I'm not too worried about the size of the response.

That being said, having a option would be great :+1:

File: test/SourceKit/VariableType/fully-qualified.swift [hunk 1]
```diff
@@ -0,0 +1,35 @@
+let x: Int = 3
```
Comment (ahoppen) [inline test/SourceKit/VariableType/fully-qualified.swift:1]: This is just a copy of `basic.swift`, right? You could also just add another `RUN:` line to it with `%FileCheck %s --check-prefix CHECK-FULLY-QUALIFIED`

**Review by ahoppen — COMMENTED** [review]
Looks good to me. One suggestion to merge two test files.

File: test/SourceKit/VariableType/fully-qualified.swift (hunk 1 above)
Comment (skrtks) [reply test/SourceKit/VariableType/fully-qualified.swift]: It is indeed, I was looking for a way to do this, but missed the custom prefix. Thanks for pointing it out!

**Review by ahoppen — APPROVED** [review]

**Comment by ahoppen** [comment]
@swift-ci Please smoke test

**Pull request merged**